made lookup of AWS IP address more generic

scalingbits opened the pull request:
# replaced ec2metadata command with lookup of IP address in EC2 meta data.
# the agent is now
# * independent on any data in the userdata field
# * the agent will wotk with multi ENIs
# * the agent is independent of the SUSE specific ec2metadat command
#
#
#  June 29, 2018, vers 1.1 , stsch@amazon.de

oalbrigt committed:
findif: improve IPv6 NIC detection

salehabbas committed:
High: jira: new resource agent

This is a new ocf ra for managing JIRA software.

salehabbas committed:
Modify indentation from tabs to spaces

salehabbas committed:
Remove the ra version

oalbrigt committed:
CTDB: fix initial probe

krig committed:
Merge pull request #1065 from oalbrigt/CTDB-fix-probe

CTDB: fix initial probe

dciabrin committed:
redis: add support for tunneling replication traffic

Add parameters in the resource agent to assign specific redis port to
each pacemaker node. When redis slave wants to connect to a redis
master, it will instead connect to a tunnel host, on the port assigned
to the targeted redis master.

This makes it possible for redis replication traffic to go through
pre-existing tunnels. This can be used to encrypt such traffic.

krig committed:
Merge pull request #904 from oalbrigt/IPsrcaddr-fix-duplicate-routes

IPsrcaddr: match exact route to avoid failing

dciabrin committed:
Fix fallback name for ocf_attribute_target

For bundles, various resource agents now use ocf_attribute_target to
get the name of the pacemaker node to store attributes on.

If a recent version of the resource agent is being run on a pacemaker
version which does not support bundles, ocf_attribute_target will
return an empty string as hostname.

Provide a fallback path so the resource agent gets a valid name when
the resource is not containerized.

salehabbas committed:
Change the way for starting and stopping actions to make them more roubst

salehabbas committed:
Remove the version attribute

krig committed:
Doc: Remove incorrect version attribute in dev guide

oalbrigt committed:
db2: improve monitor and simplify STANDBY/.../DISCONNECTED

oalbrigt committed:
Merge pull request #1066 from dciabrin/attrib-target

Fix fallback name for ocf_attribute_target

peppos committed:
Fix alter user syntax for set_mon_user_profile

after modify users on oracle, the alter user not works.
This is the error:
 ERROR: sqlplus output: alter user MONRHCS profile 'OCFMONPROFILE'#012*#012ERRORE alla riga 1:#012ORA-00931: identificativo mancante( DEFAULT )

krig committed:
Merge pull request #1069 from peppos/master

Fix alter user syntax for set_mon_user_profile

salehabbas committed:
Remove the full path of su binary

krig committed:
Merge pull request #1064 from salehabbas/master

High: jira: new resource agent

pablomh committed:
Add new possible location for CTDB_SYSCONFIG

When upgrading from Red Hat 7.3 to 7.4 the script stated that the location of the
configuration file had moved from /etc/sysconfig/ctdb to /etc/ctdb/ctdbd.conf.

krig committed:
Merge pull request #1070 from pablomh/master

CTDB: Add new possible location for CTDB_SYSCONFIG

pablomh committed:
CTDB: Cope with deprecated "idmap backend" smb.conf option 

Keep it only if we find it in the original smb.conf file and change the default
to the non-deprecated "idmap config * : backend =" format.

krig committed:
Merge pull request #1073 from vvidic/rkt2

rkt: fix possible bashisms

krig committed:
Low: IPsrcaddr: Only check for ifconfig on BSD/solaris

mgrzybek committed:
Adds RA to monitor the number of running containers/machines

mgrzybek committed:
Not in a separated branch

mgrzybek committed:
Adds the RA and update the Makefile

krig committed:
Merge pull request #1063 from oalbrigt/azure-lb

azure-lb: new resource agent

krig committed:
Merge pull request #1074 from krig/ipsrcaddr

Low: IPsrcaddr: Only check for ifconfig on BSD/solaris

krig committed:
Merge pull request #1071 from pablomh/patch-1

CTDB: Cope with deprecated "idmap backend" smb.conf option

krig committed:
Merge pull request #1068 from oalbrigt/db2-improve-monitor

db2: improve monitor and simplify STANDBY/.../DISCONNECTED

krig committed:
Merge pull request #1067 from krig/docfix

Doc: Remove incorrect version attribute in dev guide

mgrzybek committed:
Set script as executable

mgrzybek committed:
Adds LXCInfo and MachineInfo for mapage creation

krig committed:
Merge pull request #1075 from mgrzybek/containers-resource

Container-monitoring resources

krig committed:
Doc: Add naming convention to documentation

krig committed:
Merge pull request #1079 from krig/conventions

Doc: Add naming convention to documentation

krig committed:
Rename LXDInfo to lxd-info and MachineInfo to machine-info

krig committed:
Dev: Add xsltproc to dependency list

krig committed:
Dev: Fix prelude of redis agent

krig committed:
Merge pull request #1080 from krig/rename-lxc-agents

Rename LXDInfo to lxd-info and MachineInfo to machine-info

krig committed:
Low: redis: Check read bit on redis config

krig committed:
Dev: redis: Clean up redis implementation

krig committed:
Dev: Travis: Install docbook for man page generation

krig committed:
Merge pull request #1082 from krig/docbook-travis

Dev: Travis: Install docbook for man page generation

Commit:
New RA for mpathpersit

It is a new RA hanlding persist reservation for multipath
devices. It uses mpathpersist which from package multipath-tools
as backend.
fate#324044

Commit:
Add default values and makefile entries

Add default values in order to follow new RA code style.
Add makefile entries for mpathpersist.

krig committed:
Dev: Add more docbook packages to Travis

Commit:
Use default value OCF_RESKEY_*_default in metadata

Use default value OCF_RESKEY_*_default in metadata.
Name all default value OCF_RESKEY_*_default.

krig committed:
Merge pull request #1081 from krig/redis-license-etc

Fix redis license etc

Commit:
Add missing "lang=en" attribute to shortdesc

krig committed:
Merge pull request #1083 from ls-zhu/master

New RA for mpathpersist

Commit:
LVM-activate: put all validations into lvm_validate

Checking anything before reaching monitor operation is wrong.
The monitor op is always called by pacemaker to know the status
of resource. If the monitor op fails, the stop op will be called
by pacemaker to try to make sure the resource is stopped. In
LVM-activate, the vgname, activation_mode, vg_access-mode are
checked very early before any real operation. In case the vgnane
is not specified, the monitor and stop operations will fail even
thought no resource is actually running.

Signed-off-by: Eric Ren <zren@suse.com>

oalbrigt committed:
mpathpersist: fixes and cleanup

Commit:
lvmlockd: auto change lvm config to use lvmlockd

In some deployment scripts, we may not want any manual
interfere to complete the settup.

Signed-off-by: Eric Ren <zren@suse.com>

krig committed:
Merge pull request #1084 from oalbrigt/mpathpersist-fixes-and-cleanup

mpathpersist: fixes and cleanup

Commit:
Implement GTID/semi-sync based MariaDB Master/Slave resource-agent

The focus of this replication agent is to try to provide a setup
which will not lose data. This is done by tuning the semi-sync
configuration in such a way that only once data has been committed
on a slave will the master acknowledge a write.

During startup all nodes must be online and the node with the highest
GTID is used as master.

During failover it is assumed that only one node has failed and the
remaining node with the highest GTID is selected as the new master.

krig committed:
Merge pull request #929 from pyssling/master

Implement GTID based MariaDB Master/Slave resource-agent

krig committed:
Merge pull request #1085 from renzhengeek/master

Fixes for lvmlockd and LVM-activate RAs

Werkov committed:
VirtualDomain: Properly migrate VMs on node shutdown (bsc#1074014)

There are three actors in sum controlling the VMs:
- pacemaker/VirtualDomain RA
  - issuing cluster resource operations,
- libvirtd
  - executing the given commands,
- systemd and systemd-machined
  - manages scope units of running VMs.

During shutdown both libvirtd and systemd-machined termination runs
concurrently with pacemaker so it cannot properly communicate with
libvirtd and terminate the machines (e.g. stopped scope unit would turn
the VM off instead of migration that's usually intended in a HA
cluster).

We add ordering dependency both on libvirtd.service and
virt-guest-shutdown.target so that pacemaker is terminated *before* the
other mechanisms would stop the VMs. Thus it can invoke appropriate
resource action on the running VM.

In order the ordering against virt-guest-shutdown.target to work its
stop job must be in the shutdown transaction thus we start the target
when activating a VM. The target is ordering-only, it doesn't activate
any other units.

Ref: #1087

Commit:
syslog-ng: use the basename of executable to make commercial version
supported as well

gao-yan committed:
Low: vmware: Do not reference the dropped cluster property "default_action_timeout"

As of https://github.com/ClusterLabs/pacemaker/commit/a28a558f9

gao-yan committed:
Low: Xen,NodeUtilization: Use "crm_attribute --node/-N" instead of deprecated "--node-uname/-U"

As of https://github.com/ClusterLabs/pacemaker/commit/24f45dd86

gao-yan committed:
Low: sfex_daemon: Use "crm_resource --node/-N" instead of deprecated "--host-uname/-H"

As of https://github.com/ClusterLabs/pacemaker/commit/24f45dd86

gao-yan committed:
Low: db2,galera,NodeUtilization: Use "crm_attribute --quiet/-q" instead of deprecated "-Q"

As of https://github.com/ClusterLabs/pacemaker/commit/c11ce5e9b

gao-yan committed:
Low: Xen,redis: Use "crm_attribute --query/-G" instead of deprecated "--get-value"

As of https://github.com/ClusterLabs/pacemaker/commit/051c07cc2

gao-yan committed:
Low: Xen: Use "crm_attribute --name/-n" instead of deprecated "--attr-name"

As of https://github.com/ClusterLabs/pacemaker/commit/fe96d6918

oalbrigt committed:
oracle: log warning when using using sysdba instead of $MONUSR

krig committed:
Drop python from build system since nothing uses it

There's no need to require python when it isn't actually used.

krig committed:
Merge pull request #1091 from oalbrigt/oracle-monuser-log-error

oracle: log warning when using using sysdba instead of $MONUSR

krig committed:
Merge pull request #1090 from gao-yan/cli-options

Replace deprecated CLI options

krig committed:
Merge pull request #1092 from krig/unpython

Drop python from build system since nothing uses it

krig committed:
Merge pull request #1088 from abalage/master

syslog-ng: add support for commercial syslog-ng PE

ytakeshita committed:
Low: tomcat: Fix invalid stop option

"--force" is invalid option for stop. it should be "-force".

krig committed:
Merge pull request #1093 from ytakeshita/fix_invalid_option

Low: tomcat: Fix invalid stop option

Werkov committed:
Medium: Raid1: Ignore transient devices after stopping a device (bsc#1077416)

There may be a transient invalid device after we stop the MD due to
other programs processing uevent, the original device is stopped though
(`mdadm --stop` only returns when MD is stopped).

The idea suggested by Neil Brown <nfbrown@suse.com>.

Werkov committed:
Low: Raid1: remove unnecessary wait flags (bsc#1077416)

As per Neil Brown <nfbrown@suse.com>:
> In any case it is meaningless to wait for the array to become clean or
> for resync to complete after it has already been stopped.  Stopping
> the array will *always* wait for it to be clean, but will not wait for
> resync to complete - it will stop resync gracefully so that it
> continues from where it was up to when the array is restarted.

nrwahl2 committed:
This patch introduces consistency by adding an explicit trailing "s"
character to default time values for actions in resource agent metadata.
The default unit for time is seconds, but other units are supported.
Currently, some instances of time values include the explicit "s", while
others do not.

Adding the "s" will also be more consistent with the approach in
fence agents.

oalbrigt committed:
Merge pull request #1096 from nrwahl2/nrwahl2-add_sec_to_times

Low: Add "s" suffix to action default times

krig committed:
Low: ovsmonitor: Generate man page for ovsmonitor

oalbrigt committed:
Merge pull request #1099 from krig/ovsmonitor-manpage

Low: ovsmonitor: Generate man page for ovsmonitor

oalbrigt committed:
Merge pull request #1098 from dciabrin/redis-tunnel

redis: add support for tunneling replication traffic

krig committed:
Merge pull request #1095 from Werkov/fix-raid1-stop-detection

Fix raid1 stop detection

davidmhewitt committed:
Add support for cvfs

davidmhewitt committed:
Update Filesystem

davidmhewitt committed:
Skip blockdevice check

krig committed:
Merge pull request #1101 from davidmhewitt/patch-1

Add support for cvfs

Commit:
pgsql: try to create stats_temp_directory

Database fails to start correctly if the stats_temp_directory
is specified in the configuration but does not exists.

Commit:
pgsql: fix possible bashism

Quote grep regexp to prevent possible shell expansion and
fix bashism warning:

  possible bashism in pgsql line 1949 ([^] should be [!]):
    echo "$OCF_RESKEY_replication_slot_name" | grep -q -e [^a-z0-9_]

Werkov committed:
Low: Refactor systemd detection

Introduce a common helper function to express the semantics of the call
and allow easier modifications.

dud225 committed:
Fix spelling

oalbrigt committed:
awseip/awsvip: improvements (incl multi NIC support)

ZeTopHat committed:
redirect standard output/error of selinuxenabled

I noticed servers without selinux installed get spammed in the messages log by the output of that command in this script:
e.g.
lrmd[]:   notice: operation_finished: nfs-server_monitor_10000:6025:stderr [ /usr/lib/ocf/resource.d/heartbeat/nfsserver: line 183: selinuxenabled: command not found ]

oalbrigt committed:
sybaseASE: new resource agent

oalbrigt committed:
gitignore: improvements

oalbrigt committed:
awseip/awsvip: increase default "api_delay" to 3s to avoid failures

oalbrigt committed:
Merge pull request #1107 from dud225/patch-1

Fix spelling

oalbrigt committed:
Merge pull request #1110 from ZeTopHat/patch-1

redirect standard output/error of selinuxenabled

oalbrigt committed:
Merge pull request #1112 from oalbrigt/gitignore-improvements

gitignore: improvements

oalbrigt committed:
Merge pull request #1113 from oalbrigt/awseip-awsvip-increase-default-api-delay

awseip/awsvip: increase default "api_delay" to 3s to avoid failures

oalbrigt committed:
Merge pull request #1108 from oalbrigt/awseip-awsvip-improvements

awseip/awsvip: improvements (incl multi NIC support)

oalbrigt committed:
Merge pull request #1103 from vvidic/pgsql-bashism

pgsql: fix possible bashism

oalbrigt committed:
build: update ChangeLog for 4.1.1-rc1

oalbrigt committed:
Merge pull request #1115 from oalbrigt/4.1.1-rc1

build: update ChangeLog for 4.1.1-rc1

Commit:
New RA that manage IP moving in GCP

This Resource Agent uses the GCP API to move a floating IP addresse within a
GCP VPC by changing an entry in the routing table, as discussed in the
documentation :
https://cloud.google.com/solutions/best-practices-floating-ip-addresses#option_4_failover_using_routes_api_calls

This agent also configures the floating IP locally on the instance OS.

Commit:
doc: RA dev guide: fix unnecessary quotations in examples

Commit:
Merge pull request #1116 from kskmori/fix-typo-ra-dev-guide

doc: RA dev guide: fix unnecessary quotations in examples

Commit:
Merge pull request #1053 from ytakeshita/check_synchronous_standby_names

Low: pgsql: improve validations for replication mode

Commit:
pgsql: use chmod 700 for stats_temp_directory

Files inside this directory have mode 600 only so there
should be no need for 750.

oalbrigt committed:
Merge pull request #1105 from Werkov/fix-virtualdomain-migration

VirtualDomain: Properly migrate VMs on node shutdown (bsc#1074014)

Commit:
Merge pull request #1102 from vvidic/pgsql

pgsql: try to create stats_temp_directory

jszigetvari committed:
heartbeat/syslog-ng: additional parameters introduced

    * introduced resource parameters
       * control_file
       * persist_file
       * pidfile
    * normally these would have to be added to START_OPTS, but now they
      have their own parameters
    * added these in a backwards compatible way:
       * they are not mandatory
       * they have blank defaults
       * if they aren't specified, they don't show up at the command line
         at all

Signed-off-by: Janos SZIGETVARI <jszigetvari@gmail.com>

jszigetvari committed:
heartbeat/syslog-ng-pe-6: added resource file for Syslog-ng Premium Edition 6

    * added resource file for syslog-ng PE 6
    * based on the syslog-ng OSE resource file
    * modifications include
    * added resource parameters
       * qdisk_dir
       * control_file
       * persist_file
       * pidfile
    * parameters have defaults of the default Syslog-ng PE 6 install
    * these have to be changed if shared disk is used
    * added a more gentle stop method to the stop command
       * will fall back to the original method if any of the preconditions
         are missing

Signed-off-by: Janos SZIGETVARI <jszigetvari@gmail.com>

jszigetvari committed:
heartbeat/syslog-ng-pe-7: added resource file for Syslog-ng Premium Edition 7

    * added resource file for syslog-ng PE 7
    * based on the syslog-ng OSE resource file
    * modifications include
    * added resource parameters
       * control_file
       * persist_file
       * pidfile
    * parameters have defaults of the default Syslog-ng PE 7 install
    * these have to be changed if shared disk is used
    * added a more gentle stop method to the stop command
       * will fall back to the original method if any of the preconditions
         are missing

Signed-off-by: Janos SZIGETVARI <jszigetvari@gmail.com>

yiida-ha committed:
mpathpersist: Fixed problem that character string could not be correctly parsed

krig committed:
Merge pull request #1119 from yuusuke/fix_mpathpersist

mpathpersist: Fixed problem that character string could not be correc…

jszigetvari committed:
heartbeat/Makefile.am: added RAs for Syslog-ng PE 6 and 7
doc/man/Makefile.am: added RAs for Syslog-ng PE 6 and 7

    * added RAs for Syslog-ng PE 6 and 7 to the Makefile responsible for
       * installation and testing
       * manpage generation

Signed-off-by: Janos SZIGETVARI <jszigetvari@gmail.com>

oalbrigt committed:
build: update ChangeLog for 4.1.1

Werkov committed:
Low: VirtualDomain: fix warning messages in log

Changes introduced in dd9f8d47ec0dbec067442db4e4461a72aab0585e leave
behind warning messages in logs.
Pacemaker/lrmd sets umask(0026) we override that for file creation in
RA and make new directories more carefully.

nrwahl2 committed:
Medium: exportfs: Fix square bracket stripping in clientspec

This patch fixes the issue of the exportfs RA failing its monitor
operation if the clientspec matches either of two conditions:

- It includes a character class wildcard at the beginning or end of
  the string.

    Example:
      clientspec='[ab]-host.example.com'

  In this case, the leading square bracket is incorrectly stripped,
  leaving 'ab]-host.example.com'.

- It consists of an IPv6 network with a CIDR netmask.

    Example:
      clientspec='[fe80::]/64'

  In this case, the closing square bracket is incorrectly left intact,
  leaving 'fe80::]/64'.

In either of these conditions, the start operation creates the export
successfully, but the monitor immediately fails when it does not find
the string it's looking for. Similarly, the stop operation succeeds but
does not actually remove the export.

This patch replaces the sed operation that strips brackets so that:
  - We only match IPv6 addresses (using a colon to distinguish).
  - We match the closing bracket in an IPv6 network.

oalbrigt committed:
build: manpage generation without internet

oalbrigt committed:
Merge pull request #1126 from oalbrigt/build-manpage-nonet

build: manpage generation without internet

oalbrigt committed:
Merge pull request #1124 from nrwahl2/nrwahl2-fix_clientspec_strip_brackets

Medium: exportfs: Fix square bracket stripping in clientspec

nrwahl2 committed:
Low: exportfs: Fix inconsistent whitespace in exportfs_monitor

In commit 64d392d412ce9e14d4b33a70cb2180890db15a36, I introduced some
inconsistent whitespace in the exportfs_monitor function. This patch
fixes it.

krig committed:
Merge pull request #1127 from nrwahl2/nrwahl2-remove_extra_space

Low: exportfs: Fix inconsistent whitespace in exportfs_monitor

oalbrigt committed:
CI: fixes for bash path, strncpy in GCC 8 and missing docbook-style-xsl

oalbrigt committed:
Merge pull request #1129 from oalbrigt/CI-fixes

CI: fixes for bash path, strncpy in GCC 8 and missing docbook-style-xsl

stefanfriedel committed:
enable chap authentication for lio-t iscsi target implementation. if attribute 'incoming_username' is set: enable attribute 'authentication', disable attribute 'generate_node_acls' and set chap 'userid' and 'password' for each 'allowed_initiator'

oalbrigt committed:
Merge pull request #1132 from stefanfriedel/enable-chap-iscsitarget-lio-t

enable chap authentication for lio-t iscsi target implementation. if …

oalbrigt committed:
Merge pull request #1120 from dud225/gcp-vpc-move-ip

gcp-vpc-move-ip: New RA that manages IP moving in GCP

oalbrigt committed:
gcp-vpc-move-ip: fix bash path-detection

oalbrigt committed:
Merge pull request #1133 from oalbrigt/gcp-vpc-move-ip-fix-bash-detection

gcp-vpc-move-ip: fix bash path-detection

oalbrigt committed:
send_arp: dont use "-Wcast-align" due to false-positive fail on ARM.

oalbrigt committed:
Merge pull request #1134 from oalbrigt/send_arp-avoid-cast-alignment-issue-on-arm

send_arp: dont use "-Wcast-align" due to false-positive fail on ARM.

oalbrigt committed:
syslog-ng: merge Premium Edition 6 and 7 code into main agent

oalbrigt committed:
syslog-ng: added note that qdisk_dir is only supported in PE6

oalbrigt committed:
syslog-ng: fix control file issue

oalbrigt committed:
gcp-vpc-move-ip: fix manpage-generation

oalbrigt committed:
Merge pull request #1138 from oalbrigt/gcp-vpc-move-ip-fix-manpage-generation

gcp-vpc-move-ip: fix manpage-generation

oalbrigt committed:
ocf-binaries: use detected SSH-path

guessi committed:
awseip: fix allocation_id not found error

guessi committed:
awsvip: get network-id from meta-data directly

guessi committed:
awsvip: get secondary-private-ip more precisely

guessi committed:
awseip: update year of Copyright declaration

guessi committed:
awsvip: update year of Copyright declaration

oalbrigt committed:
Merge pull request #1142 from oalbrigt/ocf-binaries-use-detected-SSH-path

ocf-binaries: use detected SSH-path

guessi committed:
Revert "awseip: fix allocation_id not found error"

This reverts commit 7194265bd37ee549244eb6a8f2a5288fdf383045.

guessi committed:
awseip: fix allocation_id not found error

oalbrigt committed:
Merge pull request #1143 from guessi/improve_awsvip_awseip

Low: Improve awsvip awseip

OndrejHome committed:
log the error message in abnormal_end() function even when we are in
'stop' operation so it is clear from logs on how did we end up here

oalbrigt committed:
README: add links and IRC info

oalbrigt committed:
Merge pull request #1135 from oalbrigt/syslog-pe-6-7-merge

 syslog-ng: add Premium Edition 6 and 7 support

oalbrigt committed:
findif: replace -a with && to avoid possible issues with -a

oalbrigt committed:
Merge pull request #1145 from oalbrigt/README-add-links-and-IRC-info

README: add links and IRC info

Commit:
ldirectord: detect location of rm command

/usr/bin/rm on RHEL, /bin/rm on Debian.

mgrzybek committed:
Adds openstack resource agents

* info
* cinder-volume
* floating-ip

mgrzybek committed:
Adds the new agents to documentation

mgrzybek committed:
Adds execution rights to agents

mgrzybek committed:
Better layout for Makefile.am

mgrzybek committed:
Layout update

Uses tabs instead of spaces.
Updates readme.

mgrzybek committed:
Layout update

mgrzybek committed:
Layout updates

oalbrigt committed:
Merge pull request #1146 from vvidic/rm-path

ldirectord: detect location of rm command

mgrzybek committed:
Adds the agents to Makefile.am

mgrzybek committed:
Update openstack-info

mgrzybek committed:
Update openstack-floating-ip

mgrzybek committed:
Update openstack-floating-ip

mgrzybek committed:
Update openstack-info

Uses /bin/sh instead of /bin/bash because we dn't need it.

Commit:
ldirectord: add manpage to service documentation

Commit:
Fix spelling errors reported by lintian

oalbrigt committed:
Merge pull request #1148 from vvidic/systemd-documentation

ldirectord: add manpage to service documentation

Commit:
Squid: cleanup description for squid_port parameter

oalbrigt committed:
Merge pull request #1149 from vvidic/spelling

Fix spelling errors reported by lintian

oalbrigt committed:
Merge pull request #1121 from Werkov/fix-virtualdomain-migration

Low: VirtualDomain: fix warning messages in log

oalbrigt committed:
Merge pull request #1035 from oalbrigt/findif-improve-ipv6-nic-detection

findif: improve IPv6 NIC detection

paulmezz committed:
Volume groups and logical volumes "-" in their name get mangled with double dashes
in dmsetup.  Switching to wc and just counting lines while depending on the vgname
+ lvname filter in the dmsetup call gets around the issue with dmsetup outputting
correctly but grep failing due to the name mangle.

Logic for both test cases and dmsetup calls changed so they match too.  No reason
to not have matching tests.

This is AllBad but there isn't a better way that I'm aware of yet.

fmherschel committed:
Medium: SAPInstance: Add monitored services for ENSA2 (bsc#1092384)

For ENSA2 the S/4 HANA Enqueue Services 2 we need two strings to be
added to the default services list to be monitored: enq_server, enq_replicator.

ytakeshita committed:
Medium: pgsql: Fix incorrect SQL is selected with PostgreSQL 11 or later.

oalbrigt committed:
Merge pull request #1153 from Werkov/sapinstance-update

Medium: SAPInstance: Add monitored services for ENSA2 (bsc#1092384)

oalbrigt committed:
Merge pull request #1154 from OndrejHome/sapinstance-2018-05-1

SAPInstance improvements 2018/05 (1/4) - move logging in abnormal_end()

ytakeshita committed:
Low: pgsql: refactor OR statement from "-o" to "||"

oalbrigt committed:
Merge pull request #1159 from ytakeshita/fix_pgsql

Medium: pgsql: Fix incorrect SQL is selected with PostgreSQL 11 or later

oalbrigt committed:
Merge pull request #1147 from mgrzybek/opentack-agents

Openstack agents

Commit:
gcp-vpc-move-vip.in: manage ip alias

Add a resource agent to manage ip alias in the cluster.

start:
	Check if any machine in hostlist has the alias_ip assigned and
	disassociate it.
	Assign alias_ip to the current machine.

stop:
	Disassociate the alias_ip from the current machine.

status/monitor:
	Check if alias_ip is assigned with the current machine.

---

This is a port to the following bash script to python:
https://storage.googleapis.com/sapdeploy/pacemaker-gcp/alias

The problem with the bash script is the use of gcloud whose command line
API is not stable.

ocf-tester.in results:

	> sudo ./tools/ocf-tester.in -o alias_ip='10.128.1.0/32' -o stackdriver_logging=yes -n gcp-vpc-move-vip.in heartbeat/gcp-vpc-move-vip.in
	Beginning tests for heartbeat/gcp-vpc-move-vip.in...
	./tools/ocf-tester.in: line 226: cd: @datadir@/resource-agents: No such file or directory
	close failed in file object destructor:
	sys.excepthook is missing
	lost sys.stderr
	* rc=1: Your agent produces meta-data which does not conform to ra-api-1.dtd
	Tests failed: heartbeat/gcp-vpc-move-vip.in failed 1 tests

The only test faillig is the meta-data, but all the agents that I tried
also fails on this. If this is a concern, could you please point me out
to a test which succeeds so I can check what I am doing differently?

This commit can also be viewed at:
	https://github.com/collabora-gce/resource-agents/tree/alias

Thanks

oalbrigt committed:
gcp-vpc-move-ip: check specificed gcloud binary in validate

oalbrigt committed:
Merge pull request #1162 from oalbrigt/gcp-vpc-move-ip-fix-validate

gcp-vpc-move-ip: check specificed gcloud binary in validate

oalbrigt committed:
sybaseASE: bash path-detection

oalbrigt committed:
Merge pull request #1111 from oalbrigt/sybasease-new-ra

sybaseASE: new resource agent

oalbrigt committed:
ra-dev-guide: update instructions for GitHub

HideoYamauchi committed:
Mid: IPaddr2: Change return code.

oalbrigt committed:
Merge pull request #1168 from oalbrigt/ra-dev-guide-github-update

ra-dev-guide: update instructions for GitHub

oalbrigt committed:
Merge pull request #1169 from HideoYamauchi/change_ipaddr2

Mid: IPaddr2: Change return code.

oalbrigt committed:
Merge pull request #1151 from paulmezz/master

Volume groups and logical volumes with "-" in their name get mangled with …

Commit:
gcp-vpc-move-vip.in: minor fixes

- Get hostlist from the project if the parameter is not given
- Verify if alias is present out of each action function
- Don't call configure if 'meta-data' action is given

Commit:
gcp-vcp-move-vip.in: implement validate-all

Also fix some return errors

oalbrigt committed:
configure: add Python path detection

oalbrigt committed:
Merge pull request #1170 from oalbrigt/configure-add-python-detection

configure: add Python path detection

Commit:
gcp-vpc-move-vip.in: minor fixes

oalbrigt committed:
CI: skip Python agents in shellcheck

oalbrigt committed:
Merge pull request #1171 from oalbrigt/CI-shellsheck-skip-python-agents

CI: skip Python agents in shellcheck

oalbrigt committed:
sfex: fixes to avoid errors with latest gcc

oalbrigt committed:
Merge pull request #1172 from oalbrigt/sfex-fix-new-gcc-errors

sfex: fixes to avoid errors with latest gcc

oalbrigt committed:
Merge pull request #1167 from collabora-gce/alias

gcp-vpc-move-ip: manage ip alias

mbaldessari committed:
Do not call ocf_attribute_target in the meta-data function

Starting with pacemaker-1.1.19 a "crm_node -n" call triggers
a CRM_OP_NODE_INFO cluster operation. If this is called
from a bundle with 1.1.19 code (or later) running on a 1.1.18
cluster, during a meta-data call we will get the following error in the
cluster logs:
Jul 14 11:35:27 [20041] controller-0 crmd: error: handle_request: Unexpected request (node-info) sent to non-DC node
Jul 14 11:35:27 [20041] controller-0 crmd: error: handle_request: Unexpected <create_request_adv origin="send_node_info_request" t="crmd" version="3.0.14" subt="request" reference="node-info-crm_node-1531568127-1" crm_task="node-info" crm_sys_to="crmd" crm_sys_from="e67698a2-6f50-45fc-b8de-423c94e11c99" acl_target="root" crm_user="root" src="controller-0"/>

By not calling ocf_attribute_target (which triggers a crm_node -n
call) when polling for meta-data we do not run into this issue.

This can easily get triggered when creating a resource invoking
crm_node -n inside a 1.1.19 container with pcs, as that invokes
the 'meta-data' action explicitely.

Co-Authored-By: Damien Ciabrini <dciabrin@redhat.com>
Suggested-By: Ken Gaillot <kgaillot@redhat.com>

oalbrigt committed:
Merge pull request #1173 from mbaldessari/attribute-target-metadata

Do not call ocf_attribute_target in the meta-data function

Commit:
gcp-vpc-move-route.in: python implementation of gcp-vpc-move-ip.in

gcloud api is not reliable and it is slow, add a python version of
gcp-vpc-move-ip.in

oalbrigt committed:
Add Python library

oalbrigt committed:
gcp-vpc-move-vip: use Python library

Commit:
gcp-vpc-move-ip.in: add deprecation message

Commit:
gcp-vpc-move-vip.in: filter call to aggregatedList

Don't list all the instances in the project, filter only the one we are
interested in.

Commit:
gcp-vpc-move-route.in: move stackdriver parameter

Move stackdriver parameter to the bottom of metadata list

oalbrigt committed:
gcp-vpc-move-vip: moved alias-parameters to top of metadata

Commit:
gcp-vpc-move-route.in: minor fixes

oalbrigt committed:
Merge pull request #1176 from collabora-gce/alias

gcp-vpc-move-vip.in: filter call to aggregatedList

mbaldessari committed:
Initial podman support

Tested with the following container:
     podman container set: test_bundle [docker.io/sdelrio/docker-minimal-nginx]
       test_bundle-podman-0 (ocf::heartbeat:podman):        Started nodea
       test_bundle-podman-1 (ocf::heartbeat:podman):        Started nodeb
       test_bundle-podman-2 (ocf::heartbeat:podman):        Started nodec

Tested a couple of stop/start cycles successfully. Needs the
corresponding pacemaker support https://github.com/ClusterLabs/pacemaker/pull/1564

oalbrigt committed:
Merge pull request #1175 from collabora-gce/route

gcp-vpc-move-route.in: python implementation of gcp-vpc-move-ip.in

oalbrigt committed:
gcp-vpc-move-route: use Python library

oalbrigt committed:
Python: add logging.basicConfig() to support background logging

changfeng012008 committed:
Add Aliyun vpc-move-ip agent.

oalbrigt committed:
Merge pull request #1178 from changfeng012008/aliyun-merge

Add Aliyun vpc-move-ip agent.

oalbrigt committed:
aliyun-vpc-move-ip: fixes

oalbrigt committed:
Merge pull request #1179 from oalbrigt/aliyun-vpc-move-ip-fixes

aliyun-vpc-move-ip: fixes

oalbrigt committed:
aliyun-vpc-move-ip: fix manpage

oalbrigt committed:
Merge pull request #1180 from oalbrigt/aliyun-vpc-move-ip-fix-manpage

aliyun-vpc-move-ip: fix manpage

oalbrigt committed:
build: add missing manpages

oalbrigt committed:
Merge pull request #1182 from oalbrigt/build-add-missing-manpages

build: add missing manpages

fmherschel committed:
Merge branch 'master' of github.com:fmherschel/resource-agents

fmherschel committed:
SAPInstance: implemeted reload method
The reload method is needed to avoid resource restarts after a non-unique parameter has been changed.
This is in special for interest of the MONITOR_SERVICES parameter.

fmherschel committed:
SAPInstance: Improved indents

oalbrigt committed:
Merge pull request #1183 from fmherschel/master

 SAPInstance: implemented reload method

oalbrigt committed:
findif: only match lines containing netmasks

oalbrigt committed:
Merge pull request #1184 from oalbrigt/findif-only-match-masks

findif: only match lines containing netmasks

fmherschel committed:
SAPInstance: Improved SAP instance profile detection

Werkov committed:
aws-vpc-move-ip: Use ip utility to check address

When pinging the assigned address during initial monitor (probe) on one
node we may actually ping the reachable address when the resource is
running on another node. This yields false positive monitor result on
the pinging node. Avoid this by merely checking the assignment of the
address to an interface.

krig committed:
Merge pull request #1186 from fmherschel/master

SAPInstance: Improved SAP instance profile detection

dciabrin committed:
rabbitmq-cluster: retry start when cluster join fails

When a node tries to join an existing cluster, it fetches a node
list to try to connect from any of those running nodes.

If the nodes from this list become unavailable while we're joining
the cluster, the rabbitmq server will fail to get clustered and
make the start operation fail.

Give the resource a chance to start anyway by retrying the entire
start actions until it succeeds or until the start timeout is
reached and pacemaker stops the start operation.

Co-Authored-by: <michele@acksyn.org>
Suggested-by: <abeekhof@redhat.com>

krig committed:
Merge pull request #1188 from dciabrin/rabbitmq-retry-start

rabbitmq-cluster: retry start when cluster join fails

krig committed:
Merge pull request #1187 from Werkov/fix-aws-vpc-move-ip

RFC: aws-vpc-move-ip: Use ip utility to check address

krig committed:
Merge pull request #1177 from oalbrigt/python-library

Python library

JAORMX committed:
ipsec: Add tunnel fallback option

This option makes brings up a fallback tunnel if the main tracked tunnel
goes down. This is useful for when there is a failover in the VIP (which
is our main tunnel), and then we need to establish a tunnel to the new
node that has the VIP.

The fallback mechanism is a little bit hacky, but it's the only way I
could get it to work... basically it's put up in a background process,
which will keep trying retry putting up the fallback node. Meanwhile,
the resource agent will keep doing its job and put up the main node on
the host that received the VIP. eventually, a retry on the original node
will bring up the fallback tunnel.

oalbrigt committed:
LVM: fix missing dash

JAORMX committed:
Remove output redirection to file in /tmp

That was meant for debugging purposes.

krig committed:
Merge pull request #1190 from oalbrigt/LVM-fix-missing-dash

LVM: fix missing dash

krig committed:
Medium: mysql: Remove obsolete DEBUG_LOG functionality (bsc#1021689)

The DEBUG_LOG functionality built into the mysql agent
has temp race problems (see #1049), but it is also
deprecated by the ocf trace facility.

The simplest fix is to remove it.

krig commented:
Hm, I suspect that you wanted to make a pull request from your repository to the ClusterLabs one, that has the changes to the AWS agent? This one seems to be going in the other direction.

krig committed:
Merge pull request #1191 from krig/mysql-rm-debuglog

Medium: mysql: Remove obsolete DEBUG_LOG functionality (bsc#1021689)

oalbrigt committed:
Merge pull request #1193 from mbaldessari/podman-support

Initial podman support

oalbrigt committed:
Merge pull request #1189 from JAORMX/ipsec-update

ipsec: Add tunnel fallback option

jnpkrn committed:
spec: fix some overkil path-based Requires, trailing whitespace, thinko

...in conditional (there's no "fedora_version" macro akin
to "suse_version" convention, just mere "fedora"; introduced
with e2e76528).

Note that resource-agents got mentioned recently regarding Fedora
packaging affairs, in particular with respect to /bin/sed Requires[1].

[1] https://src.fedoraproject.org/rpms/redhat-rpm-config/pull-request/29#comment-12256

Signed-off-by: Jan Pokorný <jpokorny@redhat.com>

oalbrigt committed:
Merge pull request #1197 from jnpkrn/spec-path-requires

spec: fix some overkil path-based Requires, trailing whitespace, thinko

jeckersb committed:
rabbitmq-cluster: get cluster status from mnesia during monitor

If mnesia is not running (for example if `rabbitmqctl stop_app` has
been called, or the service has paused during partition due to the
pause_minority strategy) then the cluster_status command to
rabbitmqctl will read the cached cluster status from disk and the
command returns 0 even though the service isn't really running at all.

Instead, force the cluster status to be read from mnesia.  If mnesia
is not running due to the above or similar circumstances, the command
will catch that and properly fail the monitor action.

Resolves: RHBZ#1595753

oalbrigt committed:
Merge pull request #1202 from jeckersb/bz1595753

rabbitmq-cluster: get cluster status from mnesia during monitor
